### DISCUSSION TITLE
fully-automated-dns-and-dhcp-with-pihole-and-dnsmasq/

### DISCUSSION THREAD

**Discussion by giscus** [ironicbadger/blog.ktz.me#217] [Announcements]
# fully-automated-dns-and-dhcp-with-pihole-and-dnsmasq/

You do like declarative configurations don't you? Good! For the longest time I've made do with clicking through the Opnsense UI to add static IP reservations via MAC address mapping. It's been one of those minor friction points of network administration that I've put up with because automating the process

https://blog.ktz.me/fully-automated-dns-and-dhcp-with-pihole-and-dnsmasq/

**Comment by danielo515** [comment]
Thanks for sharing. Why the need to automate the reverse proxy? Using pihole for dhcp will give you automatic hostname resolution.

**Reply by ironicbadger** [reply]
> automatic hostname resolution

But no TLS. Caddy gives me the let's encrypt TLS certs for those domains, even local ones.

**Reply by ironicbadger** [reply]
To clarify by "local ones" I mean non publicly routable valid FQDNs like `my.local.service.domain.com`.

**Reply by danielo515** [reply]
Yes, I understand. I also use some for the https part. I don't have them automated tough, so I will have a look. Thanks

**Comment by sunjam** [comment]
Why switch away from Adguard Home?  Nothing wrong with Pihole, but I've found it offers slightly fewer features, such as dns over https.  The part from your podcasts that I'm missing is why you would actually want to change.  Maybe you said why and I didn't hear it.

Anyways, thanks for the very nice article.

**Reply by ironicbadger** [reply]
Embarrassingly I didn't really get into DoH yet despite feeling like I know I *should*. Maybe in 6 months you'll see the same post but with AGH in the title ;)

**Reply by sunjam** [reply]
So, why didn't you continue using adguard home?

**Comment by memilanuk** [comment]
I'm fairly new to the whole opnsense 'scene'... but why use the pihole for dnsmasq, when your opnsense firewall has dnsmasq and Adguard Home baked in?  What am I missing?

**Reply by ironicbadger** [reply]
Opnsense by default uses unbound and a third party plugin is required for AGH

**Reply by sunjam** [reply]
Go it, the issue is with Opnsense.  You could circumvent this by adding a secondary server, such as an old sbc, for running the adblocker.  Pihole setup makes sense since you want to run these services on the same device without the plugin.

**Comment by megabitdragon** [comment]
This is great. I was wondering if you made it work with your new network infrastructure, specifically does it work with VLANs?
